todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksRepository.java

```diff
@@ -248,7 +289,16 @@ public void call(Task task) {
                     }
                 });
 
-        return Observable.concat(localTask, remoteTask).first();
+        return Observable.concat(localTask, remoteTask).first()
+                .map(new Func1<Task, Task>() {
+                    @Override
+                    public Task call(Task task) {
+                        if (task == null) {
+                            throw new NoSuchElementException("No task found with taskId " + taskId);
```

Comment:
Would this benefit from using `flatMap()` and returning `Observable.error(new NoSuchElementException(...))` instead? See http://blog.danlew.net/2015/12/08/error-handling-in-rxjava/